larq_compute_engine/tflite/kernels/bconv2d.cc

```diff
@@ -230,25 +223,31 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
   if (output->type == kTfLiteInt8) {
     TF_LITE_ENSURE_EQ(context, output->quantization.type,
                       kTfLiteAffineQuantization);
-    TF_LITE_ENSURE_MSG(
-        context,
-        bconv2d_params->padding_type != kTfLitePaddingSame ||
-            bconv2d_params->pad_value == 1,
-        "8-bit quantization is only supported with valid or one-padding");
+
+    TF_LITE_ENSURE_MSG(context,
+                       (kernel_type == KernelType::kReference &&
+                        bconv2d_params->channels_in % 2 == 0) ||
+                           bconv2d_params->padding_type != kTfLitePaddingSame ||
+                           bconv2d_params->pad_value == 1,
+                       "8-bit quantization is only supported by the reference "
+                       "kernel, or with valid or one-padding");
```

Comment:
I'm not quite sure about this check, also because it's inside an `if int8` statement.

Maybe it should become: `if zero_padding { assert(float_output or (ref_kernel and even_c_in)) }`?

Reply:
I think the current assert is technically correct, but I agree that your way is much easier to read. I've added it in the latest commit, and it fused three asserts into one easier-to-read assert! Could you double-check that the assert logic is still correct?